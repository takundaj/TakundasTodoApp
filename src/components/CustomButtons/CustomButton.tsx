import { View, Text, TouchableOpacity } from "react-native";
import React from "react";
import styles from "./customButton.style";

type Props = {
  onPress: () => void;
  label: string;
  disabled?: boolean;
  type: "primary" | "cancel";
};

const CustomButton: React.FC<Props> = ({ label, onPress, disabled, type }) => {
  return (
    <TouchableOpacity
      style={[
        styles.container,
        {
          backgroundColor: type == "primary" ? "turquoise" : "lightgrey",
        },
      ]}
      onPress={onPress}
      disabled={disabled}
      testID="custom-button"
    >
      <Text style={styles.text}>{label}</Text>
    </TouchableOpacity>
  );
};

export default CustomButton;
