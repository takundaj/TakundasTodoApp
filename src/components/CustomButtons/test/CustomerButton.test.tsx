import { render, fireEvent } from "@testing-library/react-native";
import renderer from "react-test-renderer";
import CustomButton from "../CustomButton";

const mockProps = {
  label: "hello",
  onPress: jest.fn(),
  disabled: false,
};

test("Should render CustomButton component", () => {
  const { getByTestId } = render(
    <CustomButton {...mockProps} type="primary" />
  );

  const customButton = getByTestId("custom-button");

  expect(customButton).toBeDefined();
});

test("Should run onPress when button is pressed", () => {
  const { getByTestId } = render(
    <CustomButton {...mockProps} type="primary" />
  );

  const customButton = getByTestId("custom-button");

  fireEvent.press(customButton);

  expect(mockProps.onPress).toHaveBeenCalledWith();
});

test("Should match snapshot", () => {
  const tree = renderer.create(<CustomButton {...mockProps} type="primary" />);

  expect(tree).toMatchSnapshot();
});
