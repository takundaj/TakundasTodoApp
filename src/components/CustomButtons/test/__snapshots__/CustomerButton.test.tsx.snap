// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should match snapshot 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "turquoise",
      "borderRadius": 50,
      "marginBottom": 16,
      "opacity": 1,
      "padding": 12,
    }
  }
  testID="custom-button"
>
  <Text
    style={
      {
        "color": "white",
        "fontWeight": "bold",
      }
    }
  >
    hello
  </Text>
</View>
`;
